synthetics-stats/src/entities/incentives.ts

```diff
@@ -0,0 +1,152 @@
+    let marketInfo = getMarketInfo(marketAddress)!;
+    let timeInSeconds = event.block.timestamp.minus(BigInt.fromI32(entity.timestamp))
+    entity.cumulativeTimeByMarketTokensSupply = entity.cumulativeTimeByMarketTokensSupply.plus(
+      marketInfo.marketTokensSupply.times(timeInSeconds)
+    );
+  } else {
+    let timeInSeconds = event.block.timestamp.minus(BigInt.fromI32(entity.updatedTimestamp))
+    entity.cumulativeTimeByMarketTokensSupply = entity.cumulativeTimeByMarketTokensSupply.plus(
+      entity.lastMarketTokensSupply.times(timeInSeconds)
+    );
+  }
+  
+  entity.lastMarketTokensSupply = marketTokensSupply;
+  entity.updatedTimestamp = event.block.timestamp.toI32();
+  
+  let endTimestamp = entity.timestamp + SECONDS_IN_WEEK;
+  let extrapolatedTimeByMarketTokensSupply = entity.lastMarketTokensSupply.times(BigInt.fromI32(endTimestamp).minus(event.block.timestamp))
+  entity.weightedAverageMarketTokensSupply = entity.cumulativeTimeByMarketTokensSupply
+    .plus(extrapolatedTimeByMarketTokensSupply)
+    .div(BigInt.fromI32(SECONDS_IN_WEEK));
+  
+  entity.save();
+}
+
+function _getOrCreateLiquidityProviderIncentivesStat(account: string, marketAddress: string, period: string, timestamp: i32): LiquidityProviderIncentivesStat {
+  let startTimestamp = timestampToPeriodStart(timestamp, period);
+  let id = account + ":" + marketAddress + ":" + startTimestamp.toString();
+  let entity = LiquidityProviderIncentivesStat.load(id);
+  if (entity == null) {
+    entity = new LiquidityProviderIncentivesStat(id);
+    entity.timestamp = startTimestamp;
+    entity.period = period;
+    entity.account = account;
+
+    entity.updatedTimestamp = 0;
+    entity.lastMarketTokensBalance = ZERO;
+    entity.cumulativeTimeByMarketTokensBalance = ZERO;
+    entity.weightedAverageMarketTokensBalance = ZERO;
+  }
+  
+  return entity!;
+}
+
+function _getOrCreateMarketIncentivesStat(marketAddress: string, timestamp: i32): MarketIncentivesStat {
+  let period = "1w";
+  let startTimestamp = timestampToPeriodStart(timestamp, period);
+  let id = marketAddress + ":" + startTimestamp.toString();
+  let entity = MarketIncentivesStat.load(id);
+  
+  if (entity == null) {
+    entity = new MarketIncentivesStat(id);
+    entity.timestamp = startTimestamp;
+    entity.period = period;
+    entity.marketAddress = marketAddress;
+
+    entity.updatedTimestamp = 0;
+    entity.lastMarketTokensSupply = ZERO;
+    entity.cumulativeTimeByMarketTokensSupply = ZERO;
+    entity.weightedAverageMarketTokensSupply = ZERO;
+  }
+  
+  return entity!;
+}
+
+function _getUserMarketInfo(account: string, marketAddress: string): UserMarketInfo {
+  let id = account + ":" + marketAddress;
+  let entity = UserMarketInfo.load(id);
+
+  if (entity == null) {
+    entity = new UserMarketInfo(id);
+    entity.marketTokensBalance = ZERO;
+    entity.account = account;
+    entity.marketAddress = marketAddress;
+  }
+  
+  return entity!;
+}
+
```

Comment:
please setup prettier for project, right now I have a big diff with your changes:

<img width="1203" alt="image" src="https://github.com/gmx-io/gmx-subgraph/assets/144131435/bd7752ed-7d36-4b7d-909a-01b70dda0353">

Reply:
discussed in dm, we haven't configured prettier in project yet, will do it separately